synapse/rest/admin/users.py

```diff
@@ -626,12 +640,14 @@ class SearchUsersRestServlet(RestServlet):
 
     PATTERNS = admin_patterns("/search_users/(?P<target_user_id>[^/]*)")
 
-    def __init__(self, hs):
+    def __init__(self, hs: "HomeServer"):
         self.hs = hs
         self.store = hs.get_datastore()
         self.auth = hs.get_auth()
 
-    async def on_GET(self, request, target_user_id):
+    async def on_GET(
+        self, request: SynapseRequest, target_user_id: str
+    ) -> Tuple[int, Optional[JsonDict]]:
```

Comment:
Why is this one optional? I think `search_users` always returns a dict?

Reply:
IMO it can be `NULL` if search returns an empty result.
https://github.com/matrix-org/synapse/blob/aee10768d8a9bf6c3df80859d2c135607aa1bf80/synapse/storage/databases/main/__init__.py#L356-L372

Reply:
Ah! I was looking at the handler code. 👍  This is why adding the type hints is useful!